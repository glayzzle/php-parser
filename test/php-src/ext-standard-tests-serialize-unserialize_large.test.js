// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/standard/tests/serialize/unserialize_large.phpt
  it("Unserializing payload with unrealistically large element counts", function () {
    expect(parser.parseCode("<?php\nvar_dump(unserialize(\"a:1000000000:{}\"));\nvar_dump(unserialize(\"O:1000000000:\\\"\\\":0:{}\"));\nvar_dump(unserialize(\"O:1:\\\"X\\\":1000000000:{}\"));\nvar_dump(unserialize(\"C:1:\\\"X\\\":1000000000:{}\"));\n?>")).toMatchSnapshot();
  });
});
