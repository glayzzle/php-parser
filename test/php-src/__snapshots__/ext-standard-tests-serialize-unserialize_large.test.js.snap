// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Unserializing payload with unrealistically large element counts 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"a:1000000000:{}\\"",
                "unicode": false,
                "value": "a:1000000000:{}",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "unserialize",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"O:1000000000:\\\\\\"\\\\\\":0:{}\\"",
                "unicode": false,
                "value": "O:1000000000:\\"\\\\\\":0:{}",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "unserialize",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"O:1:\\\\\\"X\\\\\\":1000000000:{}\\"",
                "unicode": false,
                "value": "O:1:\\"X\\\\\\":1000000000:{}",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "unserialize",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Call {
            "arguments": Array [
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"C:1:\\\\\\"X\\\\\\":1000000000:{}\\"",
                "unicode": false,
                "value": "C:1:\\"X\\\\\\":1000000000:{}",
              },
            ],
            "kind": "call",
            "what": Name {
              "kind": "name",
              "name": "unserialize",
              "resolution": "uqn",
            },
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "var_dump",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
